### PR TITLE
Add CSR infra

### PR DESCRIPTION
**Pull request by mmamayka**
Adds easy configurable & awful implemented Control and Status registers

### CODE REVIEW

**Review by levBagryansky — CHANGES_REQUESTED** [review]

**Review by c71n93 — APPROVED** [review]

File: unit_test/util/bit-magic-tests.cpp [hunk 1]
```diff
@@ -83,3 +83,40 @@ TEST(ExtractBits, Dummy) {
...
+TEST(ExtractMasked, Dummy) {
+    constexpr unsigned int Mask = 0b10001110000;
+    unsigned int value = 0b11101010111;
+
+    unsigned int r = besm::util::ExtractMasked<unsigned int, Mask>(value);
+
+    EXPECT_EQ(r, 0b1101);
+}
+
+TEST(InsertMasked, Dummy) {
```
Comment (levBagryansky) [inline unit_test/util/bit-magic-tests.cpp:113]: @mmamayka please add tests to other structures, its not clear enough

**Review by levBagryansky — CHANGES_REQUESTED** [review]
@mmamayka I did not check csr headers yet

File: include/besm-666/config.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "besm-666/riscv-types.hpp"
+
+namespace besm {
+
+constexpr const RV64UDWord VENDOR_ID = 0;
+constexpr const RV64UDWord ARCH_ID = 1;
+constexpr const RV64UDWord IMP_ID = 1;
+
+enum XLEN : RV64UWord { XLEN32 = 1, XLEN64 = 2, XLEN128 = 3 };
```
[the remark this answers is not in the document]
Comment (mmamayka) [reply include/besm-666/config.hpp]: Whole file was removed due to further refactoring & development

File: include/besm-666/config.hpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+enum class Exception {
```
[the remark this answers is not in the document]
Comment (mmamayka) [reply include/besm-666/config.hpp]: Whole file was removed due to further refactoring & development

File: unit_test/util/bit-magic-tests.cpp (hunk 1 above)
Comment (mmamayka) [reply unit_test/util/bit-magic-tests.cpp:113]: Structures is a private interface of `InsertMasked` and `ExtractMasked` functions. They should not be used anywhere else.

File: include/besm-666/util/bit-magic.hpp [hunk 4]
```diff
@@ -1,11 +1,14 @@
 #pragma once
 
+#include <bitset>
 #include <cstddef>
+#include <iostream>
```
[the remark this answers is not in the document]
Comment (mmamayka) [reply include/besm-666/util/bit-magic.hpp]: Fixed

**Pull request merged**